src/main/controllers/TribunalOrderOrRequestDetailsController.ts

```diff
@@ -0,0 +1,57 @@
+import logger from '@pact-foundation/pact/src/common/logger';
+import { Response } from 'express';
+
+import { AppRequest } from '../definitions/appRequest';
+import { CLAIMANT, TranslationKeys } from '../definitions/constants';
+import { FormContent } from '../definitions/form';
+import { AnyRecord } from '../definitions/util-types';
+
+import { getDocumentAdditionalInformation } from './helpers/DocumentHelpers';
+import { getPageContent } from './helpers/FormHelpers';
+import { getLanguageParam } from './helpers/RouterHelpers';
+import { getRepondentOrderOrRequestDetails } from './helpers/TribunalOrderOrRequestHelper';
+
+export default class TribunalOrderOrRequestDetailsController {
+  public get = async (req: AppRequest, res: Response): Promise<void> => {
+    const userCase = req.session.userCase;
+    const selectedRequestOrOrder = userCase.sendNotificationCollection.find(it => it.id === req.params.orderId);
+
+    userCase.selectedRequestOrOrder = selectedRequestOrOrder;
+
+    //todo URL must be reviewed and changed to 'Respond to request'.
+    const redirectUrl = `/respondent-order-or-request-details/${selectedRequestOrOrder.value.number}${getLanguageParam(
+      req.url
+    )}`;
+    const respondButton =
+      selectedRequestOrOrder.value.respondCollection?.filter(r => r.value.from === CLAIMANT).length === 0;
```

Comment:
This can still be shortened, same comment as before. You just need to check if !any(r => r.value.from...)

Reply:
I changed it  but in my opinion every NOT symbol reduces the readability of the code:)